contracts/HATVault.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+// Disclaimer https://github.com/hats-finance/hats-contracts/blob/main/DISCLAIMER.md
+
+pragma solidity 0.8.14;
+
+
+import "./vaults/Claim.sol";
+import "./vaults/Deposits.sol";
+import "./vaults/Params.sol";
+import "./vaults/Withdrawals.sol";
+
+contract HATVault is Claim, Deposits, Params, Withdrawals {
```

Comment:
Can we inherit from some kind of pre-exisitng erc4626 interface?

Reply:
We do https://github.com/hats-finance/hats-contracts/pull/276/files#diff-5ff9b4e6ce18e39f2814e7d6b168516c5ee11e9f3057624fe4f06c33281a6582R82

Again, much nicer as a single file imo

Reply:
nvm it's in base :)